mm2src/coins/eth.rs

```diff
@@ -6385,6 +6464,11 @@ pub async fn eth_coin_from_conf_and_request(
         erc20_tokens_infos: Default::default(),
         nfts_infos: Default::default(),
         platform_fee_estimator_state,
+        gas_limit: if conf["gas_limit"].is_null() {
+            Default::default()
+        } else {
+            json::from_value(conf["gas_limit"].clone()).map_err(|e| e.to_string())?
+        },
```

Comment:
You have code duplication related to "gas_limit" value creation.
Please move logic to priv function

```rust
fn extract_gas_limit(conf: &Json) -> Result<EthGasLimitOverride, String> {
    if conf["gas_limit"].is_null() {
        Ok(Default::default())
    } else {
        json::from_value(conf["gas_limit"].clone()).map_err(|e| e.to_string())
    }
}
```

```rust
        nfts_infos: Default::default(),
        platform_fee_estimator_state,
        gas_limit: extract_gas_limit(conf)?,
        abortable_system,
    };

```

Reply:
fixed in [3d056cd](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2137/commits/3d056cd92dd0ab2ff6b9ecba980fa486f5ddaf8b)
Thanks!